rasa/cli/data.py

```diff
@@ -182,3 +207,86 @@ def _validate_story_structure(validator: Validator, args: argparse.Namespace) ->
     return validator.verify_story_structure(
         not args.fail_on_warnings, max_history=args.max_history
     )
+
+
+def _convert_nlu_data(args: argparse.Namespace):
+    if args.format in ["json", "md"]:
+        convert_training_data(args.data, args.out, args.format, args.language)
+    elif args.format == "yaml":
+        convert_to_yaml(args, True)
+    else:
+        print_error_and_exit(
+            "Did not recognize output format. Supported output formats: 'json', "
+            "'md', 'yaml'. Specify the desired output format with '--format'."
+        )
+
+
+def _convert_core_data(args: argparse.Namespace):
+    if args.format == "yaml":
+        convert_to_yaml(args, False)
+    else:
+        print_error_and_exit(
+            "Did not recognize output format. Supported output formats: "
+            "'yaml'. Specify the desired output format with '--format'."
+        )
+
+
+def convert_to_yaml(args: argparse.Namespace, is_nlu: bool) -> None:
+
+    output = Path(args.out)
+    if not os.path.exists(output):
+        print_error_and_exit(
+            f"The output path '{output}' doesn't exist. Please make sure to specify "
+            f"an existing directory and try again."
+        )
+
+    training_data = Path(args.data)
+    if not os.path.exists(training_data):
+        print_error_and_exit(
+            f"The training data path {training_data} doesn't exist "
+            f"and will be skipped."
+        )
+
+    num_of_files_converted = 0
+    for file in os.listdir(training_data):
+        source_path = training_data / file
+        output_path = output / f"{source_path.stem}{CONVERTED_FILE_SUFFIX}"
+
+        if MarkdownReader.is_markdown_nlu_file(source_path):
+            if not is_nlu:
+                continue
+            _write_nlu_yaml(source_path, output_path, source_path)
+            num_of_files_converted += 1
+        elif not is_nlu and MarkdownStoryReader.is_markdown_story_file(source_path):
+            _write_core_yaml(source_path, output_path, source_path)
+            num_of_files_converted += 1
+        else:
+            print_warning(f"Skipped file '{source_path}'")
```

Comment:
The warning is not showing when doing `rasa convert core` but iterating over NLU files. Maybe the `if` structure can be changed like this:

```suggestion
        if is_nlu and MarkdownReader.is_markdown_nlu_file(source_path):
            _write_nlu_yaml(source_path, output_path, source_path)
            num_of_files_converted += 1
            continue
 
        if not is_nlu and MarkdownStoryReader.is_markdown_story_file(source_path):
            _write_core_yaml(source_path, output_path, source_path)
            num_of_files_converted += 1
            continue

        print_warning(f"Skipped file '{source_path}'")
```

Reply:
It's a little bit trickier. `MarkdownStoryReader.is_markdown_story_file` returns `true` for the `NLU` files 🤦 
So our `rasa train` and other commands work only because we first check if it's `NLU` before `Core`.
So this `if` condition is hacky but correct.